react-stopwatch/my-app/src/StopWatch.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react';
+import { FaPause, FaPlay } from 'react-icons/fa';
+
+export function StopWatch() {
+  const [elapsedTime, setElapsedTime] = useState(0);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [intervalId, setIntervalId] = useState<NodeJS.Timeout>();
+
+  function handleToggle() {
+    if (!isPlaying) {
+      const id = setInterval(() => {
+        setElapsedTime((prev) => prev + 1);
+      }, 1000);
+      setIntervalId(id);
+      setIsPlaying(true);
+    } else {
+      if (intervalId !== undefined) {
+        clearInterval(intervalId);
+        setIntervalId(undefined);
+      }
+      setIsPlaying(false);
+    }
+  }
```

Comment:
As we discussed in the Q&A, it would probably be better to break this up into multiple functions. One for just playing, and one for just pausing. It's good to have your functions only do one thing. Either handle all the logic for playing, or all the logic for pausing.